staging/src/k8s.io/kubectl/pkg/cmd/wait/wait.go

```diff
@@ -152,6 +156,7 @@ func (flags *WaitFlags) AddFlags(cmd *cobra.Command) {
 
 	cmd.Flags().DurationVar(&flags.Timeout, "timeout", flags.Timeout, "The length of time to wait before giving up.  Zero means check once and don't wait, negative means wait for a week.")
 	cmd.Flags().StringVar(&flags.ForCondition, "for", flags.ForCondition, "The condition to wait on: [delete|condition=condition-name[=condition-value]|jsonpath='{JSONPath expression}'=[JSONPath value]]. The default condition-value is true.  Condition values are compared after Unicode simple case folding, which is a more general form of case-insensitivity.")
+	cmd.Flags().BoolVar(&flags.WaitForCreation, "wait-for-creation", flags.WaitForCreation, "If set to true, also wait for creation of objects if they do not already exist. This flag is ignored in --for=delete")
```

Comment:
Rather than documenting it is ignored with `--for=delete`, can we say it cannot be used with `--for=delete` and validate / reject that case? Similarly, document / validate it cannot be used with `--timeout=0`?

Reply:
Sounds good is it's default is false now. Added rejections and updated the document.